src/lib/hooks/useWindowSize.ts

```diff
@@ -0,0 +1,30 @@
+import { useEffect, useState } from 'react';
+
+// taken from https://usehooks.com/useWindowSize/
+export const useWindowSize = () => {
+  // Initialize state with undefined width/height so server and client renders match
+  // Learn more here: https://joshwcomeau.com/react/the-perils-of-rehydration/
+  const [windowSize, setWindowSize] = useState({});
+
+  useEffect(() => {
+    // Handler to call on window resize
+    function handleResize() {
+      // Set window width/height to state
+      setWindowSize({
+        width: window.innerWidth,
+        height: window.innerHeight
+      });
+    }
```

Comment:
This doesn't need to be inside the `useEffect` right? Seems like we should pull it up to the outer function scope.